### PR TITLE
Report the correct field and fix typo in log messages.

### PR DESCRIPTION
**Pull request by kishen-v**
Fixes: https://github.com/kubernetes/test-infra/issues/27847
  
The `failed_context_names` field incorrectly reports the `context_names` in the log messages. This change is made to report the correct field in the log message.

### CODE REVIEW

File: prow/tide/tide.go [hunk 1]
```diff
@@ -803,7 +803,7 @@ func unsuccessfulContexts(contexts []Context, cc contextChecker, log *logrus.Ent
 		"total_context_count":  len(contexts),
 		"context_names":        contextsToStrings(contexts),
 		"failed_context_count": len(failed),
-		"failed_context_names": contextsToStrings(contexts),
+		"failed_context_names": contextsToStrings(failed),
```
Comment (chaodaiG) [inline prow/tide/tide.go:806]: https://maxchadwick.xyz/blog/testing-log-output-in-go-logrus, a unit test would be appreciated

File: prow/tide/tide_test.go [hunk 2]
```diff
@@ -2820,6 +2842,8 @@ func TestIsPassing(t *testing.T) {
 				OptionalContexts:    []string{"c4"},
 				SkipUnknownContexts: &yes,
 			},
+			availableContexts: []string{"c1", "c2", "c3", "c4", "c5"},
+			failedContexts:    []string(nil),
```
Comment (Karthik-K-N) [inline prow/tide/tide_test.go:2846]: I think If its nil slice we can remove this line.

File: prow/tide/tide.go [hunk 3]
```diff
@@ -457,6 +457,7 @@ func contextsToStrings(contexts []Context) []string {
 	for _, c := range contexts {
 		names = append(names, string(c.Context))
 	}
+	sort.Strings(names)
```
Comment (Karthik-K-N) [inline prow/tide/tide.go:460]: Why sorting here, Is that because it helps in UT?

Comment (kishen-v) [reply prow/tide/tide.go:460]: The contexts are sorted for the following reasons:
1. Helps in UT. 
2. Improves readability in logs as the contexts are sorted.

File: prow/tide/tide_test.go (hunk 2 above)
Comment (kishen-v) [reply prow/tide/tide_test.go]: Addressed!

File: prow/tide/tide.go (hunk 1 above)
Comment (kishen-v) [reply prow/tide/tide.go:807]: Added supporting UT for the changes.

**Review by petr-muller — APPROVED** [review]
/hold
LGTM and holding for @chaodaiG to have a chance to see his review addressed

File: prow/tide/tide.go (hunk 3 above)
Comment (chaodaiG) [reply prow/tide/tide.go:460]: please add this as comment

File: prow/tide/tide.go (hunk 1 above)
Comment (chaodaiG) [reply prow/tide/tide.go:807]: cool

**Review by chaodaiG — APPROVED** [review]

File: prow/tide/tide.go (hunk 3 above)
Comment (kishen-v) [reply prow/tide/tide.go:461]: Added comment as suggested.

**Pull request merged**